src/fhirtypes/ElementDefinition.ts

```diff
@@ -947,6 +960,21 @@ export class ElementDefinition {
     const targetType = this.getTargetType(target, fisher);
     const targetTypes: ElementDefinitionType[] = targetType ? [targetType] : this.type;
 
+    // If the target type is a CodeableReference but the rule types were set via the Reference() keyword,
+    // log a warning to use CodeableReference keyword
+    if (
+      targetTypes.some(t => t.code === 'CodeableReference') &&
+      !targetTypes.some(t => t.code === 'Reference') &&
+      rule.types.every(t => t.isReference)
+    ) {
```

Comment:
Ah, that's a good point, and a case I hadn't thought about. If I just change the condition here, the warning message actually constructs the wrong example because it tries to create `CodeableReference(boolean or Patient)`, which is wrong. So I think I'll also just remove the example, especially since the line number is included as well.

Reply:
I updated the condition and the warning message and included a test in [20614c4](https://github.com/FHIR/sushi/pull/1292/commits/20614c424ca05afcf4c971a81cf8eb387c4a11c3).